changes.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package bundlechanges
+
+import (
+	"fmt"
+	"sort"
+
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// FromData generates and returns the set of changes required to deploy the
+// given bundle data.
+func FromData(data *charm.BundleData) []*Change {
+	cs := &changeset{}
+	addedServices := handleServices(cs.add, data.Services)
+	addedMachines := handleMachines(cs.add, data.Machines)
+	handleRelations(cs.add, data.Relations, addedServices)
+	handleUnits(cs.add, data.Services, addedServices, addedMachines)
+	return cs.changes
+}
+
+// Change holds a single change required to deploy a bundle.
+type Change struct {
+	// Id is the unique identifier for this change.
+	Id string `json:"id"`
+	// Method is the action to be performed to apply this change.
+	Method string `json:"method"`
+	// Args holds a list of arguments to pass to the method.
+	Args []interface{} `json:"args"`
+	// Requires holds a list of dependencies for this change. Each dependency
+	// is represented by the corresponding change id, and must be applied
+	// before this change is applied.
+	Requires []string `json:"requires"`
+}
+
+// changeset holds the list of changes returned by FromData.
+type changeset struct {
+	changes []*Change
+}
+
+// add is an addChangeFunc that can be used to add a change to this change set.
+func (cs *changeset) add(method string, requires []string, args ...interface{}) *Change {
+	if args == nil {
```

Comment:
The Python version of the library returns an empty list when there are no args, so it does not omit the args, and it does not return null. So the contract is to return an empty list and this is the only method I know to do that in Go. Is there another way?